cpp/bench/sg/svc.cu

```diff
@@ -0,0 +1,129 @@
+#include "cuml/svm/svm_model.h"
+#include "cuml/svm/svm_parameter.h"
+
+namespace ML {
+namespace Bench {
+namespace SVM {
+
+template <typename D>
+struct Params {
+  DatasetParams data;
+  BlobsParams blobs;
+  MLCommon::Matrix::KernelParams kernel;
+  ML::SVM::svmParameter svm_param;
+  ML::SVM::svmModel<D> model;
+};
+
+template <typename D>
+class SVC : public BlobsFixture<D, D> {
+ public:
+  SVC(const std::string& name, const Params<D>& p)
+    : BlobsFixture<D, D>(p.data, p.blobs),
+      kernel(p.kernel),
+      model(p.model),
+      svm_param(p.svm_param) {
+    std::vector<std::string> kernel_names{"linear", "poly", "rbf", "tanh"};
+    std::ostringstream oss;
+    oss << name << "/" << kernel_names[kernel.kernel] << p.data;
+    this->SetName(oss.str().c_str());
+  }
+
+ protected:
+  void runBenchmark(::benchmark::State& state) override {
+    if (this->params.rowMajor) {
+      state.SkipWithError("SVC only supports col-major inputs");
+    }
+    if (this->svm_param.svmType != ML::SVM::C_SVC) {
+      state.SkipWithError("SVC currently only supports C_SVC");
+    }
+    auto& handle = *this->handle;
+    auto stream = handle.getStream();
+    for (auto _ : state) {
+      CudaEventTimer timer(handle, state, true, stream);
+      ML::SVM::svcFit(handle, this->data.X, this->params.nrows,
+                      this->params.ncols, this->data.y, this->svm_param,
+                      this->kernel, this->model);
+      CUDA_CHECK(cudaStreamSynchronize(stream));
+      ML::SVM::svmFreeBuffers(handle, this->model);
+    }
+  }
+
+ private:
+  MLCommon::Matrix::KernelParams kernel;
+  ML::SVM::svmParameter svm_param;
+  ML::SVM::svmModel<D> model;
+};
+
+template <typename D>
+std::vector<Params<D>> getInputs() {
+  struct Triplets {
+    int nrows, ncols, nclasses;
+  };
+  std::vector<Params<D>> out;
+  Params<D> p;
+
+  p.data.rowMajor = false;
+
+  p.blobs.cluster_std = 1.0;
+  p.blobs.shuffle = false;
+  p.blobs.center_box_min = -2.0;
+  p.blobs.center_box_max = 2.0;
+  p.blobs.seed = 12345ULL;
+
+  //svmParameter{C, cache_size, max_iter, nochange_steps, tol, verbosity})
+  p.svm_param = ML::SVM::svmParameter{
+    1, 200, 100, 100, 1e-3, CUML_LEVEL_INFO, 0, ML::SVM::C_SVC};
+  p.model =
+    ML::SVM::svmModel<D>{0, 0, 0, nullptr, nullptr, nullptr, 0, nullptr};
+
+  std::vector<Triplets> rowcols = {
+    {5000, 2, 2}, {100, 10000, 2}, {1000, 1000, 2}};
+
+  std::vector<MLCommon::Matrix::KernelParams> kernels{
+    MLCommon::Matrix::KernelParams{MLCommon::Matrix::LINEAR, 3, 1, 0},
+    MLCommon::Matrix::KernelParams{MLCommon::Matrix::POLYNOMIAL, 3, 1, 0},
+    MLCommon::Matrix::KernelParams{MLCommon::Matrix::RBF, 3, 1, 0},
+    MLCommon::Matrix::KernelParams{MLCommon::Matrix::TANH, 3, 0.1, 0}};
+
+  for (auto& rc : rowcols) {
+    p.data.nrows = rc.nrows;
+    p.data.ncols = rc.ncols;
+    p.data.nclasses = rc.nclasses;
+    for (auto kernel : kernels) {
+      p.kernel = kernel;
+      p.kernel.gamma = 1 / rc.ncols;
```

Comment:
safer to make this `p.kernel.gamma = 1.0 / rc.ncols;` to avoid integer division, right?

Reply:
that is right, fixed it